packages/compiler/src/parser.ts

```diff
@@ -9,7 +9,10 @@ export function parseChildComponentPaths(transpiledWidget: string) {
       const source = match.groups?.src;
       widgetInstances[source] = {
         source,
-        transform: (widgetSource: string, widgetComponentName: string) => widgetSource.replaceAll(match[0], match[0].replace('Widget', widgetComponentName)),
+        transform: (widgetSource: string, widgetComponentName: string) => {
+          const signaturePrefix = `${widgetComponentName},{__bweMeta:{parentMeta:props.__bweMeta},`;
```

Comment:
Injecting this into each child Component's call site is how `parentMeta` is chained along. This is not a great way to do it, might be cleaner to modify the AST directly.